adbc.h

```diff
@@ -746,6 +746,22 @@ AdbcStatusCode AdbcStatementBindStream(struct AdbcStatement* statement,
                                        struct ArrowArrayStream* values,
                                        struct AdbcError* error);
 
+/// \brief Get the schema for bound parameters.
+///
+/// This should be called after AdbcStatementPrepare.  This retrieves
+/// an Arrow schema describing the number, names, and types of the
+/// parameters in a parameterized statement.  Not all drivers will
+/// support this.  If the name of a parameter cannot be determined,
+/// the name of the corresponding field in the schema will be an empty
+/// string.  Similarly, if the type cannot be statically determined,
+/// the type of the corresponding field will be NA (NullType).
```

Comment:
should we also explicitly state/define that the order of the columns in the schema should match the ordinal position of the parameters and if a named parameter is used multiple times in the query, it should only appear once in the schema?

Reply:
Good idea - updated the docstrings